protobufs/livekit_sip.proto

```diff
@@ -320,29 +331,35 @@ message SIPDispatchRule {
 }
 
 message CreateSIPDispatchRuleRequest {
-  SIPDispatchRule rule = 1;
+  SIPDispatchRuleInfo dispatch_rule = 8; // Rule ID is ignored
```

Comment:
This change is moving to a different API pattern to what we've used in other APIs (outside SIP at least). Can you describe the motivation?

Reply:
If you mean creation of objects by providing the full proto - the main motivation is simplicity.

Using separate fields in `Create*Request` and the actual object requires updating the code to pass these fields from one struct to the other with no additional benefit.

At least in SIP, all objects are  config-like and are saved as-is. There's no additional transformation done on them, thus I think there's no point in having two separate messages.

Reply:
The motivation for separate Create*Request messages has been that some of the fields in the *Info objects are usually not user settable (IDs, status, state, etc). As the Info objects become larger, it can become non trivial to understand which field can be set and which can't. 

I'm not necessarily against this change, but it seems like we are facing our customers and ourselves through an API change to a version that also has drawbacks. Do we have a good reason to do it? At the very least, I'd feel more conformable with this change if we can convince ourselves that we'll never add "state" related fields to the info (ie, "created_d/updated_at/dispatch_list/participant_list")

Reply:
OK, I understand now why you are concerned about this change.

Unfortunately, it's a bit too late to fix it - inbound and outbound trunk APIs were already implemented this way. They assume all fields are static. So this change was only to make dispatch rules consistent with newer trunk API (and more maintainable).

I think we can guarantee that this API version will never contain `state`, timestamps or any other dynamic fields.

Instead, I'd propose to define a new API that properly separates dynamic and static fields into separate sub-messages, when needed. I have a draft of it already locally, but it's only an experiment for now.

So to make it specific, I'll add a comment here stating that only static fields are allowed in this message.

Reply:
I think requiring all of the fields will also make it difficult to only update a specific attribute.. For example, if I'm only looking to update the numbers field, it seems weird to have to include all of the other attributes as well.

I think it's too late to change the `Create*` APIs, which also doesn't suffer from this issue, we should decide what we want to do for Update before moving forward

Reply:
@davidzhao please see discussion in https://github.com/livekit/livekit/pull/3141#issuecomment-2441779144